Analysis/HistProducerFile.py

```diff
@@ -0,0 +1,183 @@
+
+
+    print(f"time to get all dataframes dict is {datetime.now() - startTime}")
+    all_histograms = GetHistogramDictFromDataframes(all_dataframes, hist_cfg_dict,dataset)
+    print(f"time to get all histograms dict is {datetime.now() - startTime}")
+    return all_histograms
+
+
+if __name__ == "__main__":
+    import argparse
+    import yaml
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--inFile', required=True, type=str)
+    parser.add_argument('--outDir', required=False, type=str)
+    parser.add_argument('--dataset', required=True, type=str)
+    parser.add_argument('--test', required=False, type=bool, default=False)
+    parser.add_argument('--deepTauVersion', required=False, type=str, default='v2p1')
+    parser.add_argument('--compute_unc_variations', type=bool, default=False)
+    parser.add_argument('--compute_rel_weights', type=bool, default=False)
+    parser.add_argument('--histConfig', required=True, type=str)
+    parser.add_argument('--furtherCut', required=False, type=str, default = "")
+    args = parser.parse_args()
+    startTime = datetime.now()
+    headers_dir = os.path.dirname(os.path.abspath(__file__))
+    ROOT.gROOT.ProcessLine(f".include {os.environ['ANALYSIS_PATH']}")
+    ROOT.gInterpreter.Declare(f'#include "include/HistHelper.h"')
+    ROOT.gInterpreter.Declare(f'#include "include/Utilities.h"')
+    if not os.path.isdir(args.outDir):
+        os.makedirs(args.outDir)
+
+    inFile_idx_list = args.inFile.split('/')[-1].split('.')[0].split('_')
+    inFile_idx = inFile_idx_list[1] if len(inFile_idx_list)>1 else 0
+    hist_cfg_dict = {}
+    unc_cfg_dict = {}
+    unc_cfg = os.path.join(os.environ['ANALYSIS_PATH'],"config/weight_definition.yaml")
+    with open(unc_cfg, 'r') as f:
+        unc_cfg_dict = yaml.safe_load(f)
+    with open(args.histConfig, 'r') as f:
+        hist_cfg_dict = yaml.safe_load(f)
+    vars_to_plot = list(hist_cfg_dict.keys())
+    sample_cfg_dict = {}
+    sample_cfg = os.path.join(os.environ['ANALYSIS_PATH'],"config/samples_Run2_2018.yaml")
+    with open(sample_cfg, 'r') as f:
+        sample_cfg_dict = yaml.safe_load(f)
+
+
+    #if args.test: print(f"Running on file {args.inFile}")
+    all_histograms = GetHistograms(args.inFile, args.dataset, unc_cfg_dict, sample_cfg_dict, hist_cfg_dict,
+                                       args.deepTauVersion, args.compute_unc_variations, args.compute_rel_weights)
+
+    for var in all_histograms.keys():
+        finalDir = os.path.join(args.outDir, var)
+        if not os.path.isdir(finalDir):
+            os.makedirs(finalDir)
+        finalFile = ROOT.TFile(f'{finalDir}/tmp_{args.dataset}_{inFile_idx}.root','RECREATE')
+        print(f"the final file name will be {finalDir}/tmp_{args.dataset}_{inFile_idx}.root")
+        SaveHists(all_histograms[var], finalFile)
+        finalFile.Close()
+    print(f"the script took {datetime.now() - startTime}")
+
```

Comment:
Move `{finalDir}/tmp_{args.dataset}_{inFile_idx}.root` to `{finalDir}/{args.dataset}_{inFile_idx}.root`

Reply:
ok